runners/direct-java/src/main/java/org/apache/beam/runners/direct/WatermarkManager.java

```diff
@@ -834,21 +842,42 @@ public void updateWatermarks(
         timerUpdate,
         result,
         earliestHold));
+    tryApplyPendingUpdates();
+  }
+
+  private void tryApplyPendingUpdates() {
+    if (refreshLock.tryLock()) {
+      try {
+        applyNUpdates(10);
```

Comment:
Break out magic number into a constant. Also `tryApplyPendingUpdates` and `applyPendingUpdates` sound like they are analogous, but in fact they differ not only by the "try" vs "definitely do" aspect, but also by the _n_.